pkg/client/client_venconn.go

```diff
@@ -0,0 +1,176 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"github.com/jetstack/preflight/api"
+	"github.com/jetstack/preflight/pkg/version"
+	venapi "github.com/jetstack/venafi-connection-lib/api/v1alpha1"
+	"github.com/jetstack/venafi-connection-lib/venafi_client"
+	"github.com/jetstack/venafi-connection-lib/venafi_client/auth"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+type VenConnClient struct {
+	baseURL          string
+	agentMetadata    *api.AgentMetadata
+	connHandler      venafi_client.ConnectionHandler
+	installNamespace string       // Namespace in which the agent is running in.
+	venConnName      string       // Name of the VenafiConnection resource to use.
+	client           *http.Client // Used to make HTTP requests to Venafi Cloud.
+}
+
+// NewVenConnClient returns a new instance of the VenConnClient type that will perform HTTP requests using
+// no authentication.
+func NewVenConnClient(c *http.Client, agentMetadata *api.AgentMetadata, baseURL, kubeconfigPath, installNamespace, venConnName string) (*VenConnClient, error) {
+	cfg, err := loadRESTConfig(kubeconfigPath)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	cfg.Impersonate = rest.ImpersonationConfig{
+		UserName: fmt.Sprintf("system:serviceaccount:%s:venafi-connection", installNamespace),
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	clientset.DiscoveryClient.RESTClient()
```

Comment:
These lines are not used.

Reply:
Well caught, another leftover from a previous attempt. I'll remove that.